File: .scratch-theme-test/sample.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

type Task struct {
	ID     int    `json:"id"`
	Name   string `json:"name"`
	Status string `json:"status"`
}

func loadTasks(path string) ([]Task, error) {
	raw, err := os.ReadFile(path)
	if err != nil {
		return nil, err
	}
	var tasks []Task
	if err := json.Unmarshal(raw, &tasks); err != nil {
		return nil, err
	}
	return tasks, nil
}

func main() {
	tasks := []Task{{ID: 1, Name: "build", Status: "idle"}}
	for _, t := range tasks {
		fmt.Printf("running %s\n", t.Name)
	}
}

File: .scratch-theme-test/sample.py
import json
from dataclasses import dataclass


@dataclass
class Task:
    id: int
    name: str
    status: str = "idle"


def load_tasks(path: str) -> list[Task]:
    with open(path, encoding="utf-8") as fh:
        data = json.load(fh)
    return [Task(**item) for item in data if item["status"] != "done"]


class Scheduler:
    def __init__(self) -> None:
        self.tasks: list[Task] = []

    def add(self, task: Task) -> None:
        self.tasks.append(task)

    def run(self) -> int:
        count = 0
        for task in self.tasks:
            print(f"running {task.name}")
            count += 1
        return count


if __name__ == "__main__":
    scheduler = Scheduler()
    scheduler.add(Task(1, "build"))
    print(scheduler.run())
